### PR TITLE
Use agoric's synchetic chain image

### PR DESCRIPTION
**Pull request by toliaqat**
This PR utilizes the Agoric chain's mainnet synthetic chain Docker image from the Agoric-3-proposals repo, rather than replicating the Agoric chain build process. This image includes all the upgrade proposals made to the agoric mainnet.

Testing was conducted using the following commands.
```
./heighliner build -c agoric -g main
docker run -it --rm --entrypoint sh agoric:main
```

### CODE REVIEW

File: dockerfile/agoric/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM ghcr.io/strangelove-ventures/infra-toolkit:v0.1.0 AS infra-toolkit
+RUN addgroup --gid 1025 -S heighliner && adduser --uid 1025 -S heighliner -G heighliner
+
+FROM ghcr.io/agoric/agoric-3-proposals:main
```
Comment (ivanlei) [inline dockerfile/agoric/Dockerfile:4]: @toliaqat - could this be turned into an `ARG AGORIC_BASE_IMAGE` such that someone could run something akin to:
```
heighliner build -c agoric -g main --build-env 'AGORIC_BASE_IMAGE="ghcr.io/agoric/agoric-3-proposals:myprivatetag"
```

**Review by ivanlei — CHANGES_REQUESTED** [review]
I've been able to use this successfully but needed to hand edit the agoric docker image to build from

**Comment by agouin** [comment]
Thanks for the PR @toliaqat ! We already have the `imported` dockerfile for this purpose however, which enables building on top of existing docker repository images. We actually built with initially for agoric before we switched to build agoric from source. I think it would be better to switch agoric back to the imported dockerfile and extend the imported dockerfile if necessary than bring in an agoric specific dockerfile.

**Review by mrzigha — APPROVED** [review]

**Comment by nourspace** [comment]
If still needed please resubmit with new chain files format.

**Pull request closed without merging**